File: src/main/java/org/panteleyev/jpackage/JPackageGradlePlugin.java
/*
 Copyright © 2021-2025 Petr Panteleyev <petr@panteleyev.org>
 SPDX-License-Identifier: BSD-2-Clause
 */
package org.panteleyev.jpackage;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

/**
 * JPackage Gradle Plugin.
 */
public class JPackageGradlePlugin implements Plugin<Project> {
    private static final String GROUP = "Distribution";
    private static final String DESCRIPTION = "Creates application bundle using jpackage.";

    @Override
    public void apply(Project target) {
        target.getTasks().register("jpackage", JPackageTask.class, task -> {
            task.setGroup(GROUP);
            task.setDescription(DESCRIPTION);
        });
    }
}

File: src/main/kotlin/org/panteleyev/jpackage/JPackageGradlePlugin.kt
/*
 Copyright (c) Petr Panteleyev. All rights reserved.
 Licensed under the BSD license. See LICENSE file in the project root for full license information.
 */
package org.panteleyev.jpackage

import org.gradle.api.Project
import org.gradle.api.Plugin

/**
 * JPackage Gradle plugin.
 */
class JPackageGradlePlugin: Plugin<Project> {
    override fun apply(project: Project) {
        project.tasks.register("jpackage", JPackageTask::class.java);
    }
}
